pkgs/by-name/li/limine/package.nix

```diff
@@ -0,0 +1,60 @@
+# Builds limine with all available features.
+
+{
+  # Helpers
+  clangStdenv
+, fetchurl
+, lib
+, # Dependencies
+  llvmPackages
+, mtools
+, nasm
+}:
+
+let
+  version = "7.1.0";
+in
+clangStdenv.mkDerivation {
+  inherit version;
+  pname = "limine";
+  # We don't use the Git source but the release tarball, as the source has a
+  # `./bootstrap` script performing non-standard network access to download
+  # resources (a custom Git submodule-like replacement). Packaging this in Nix
+  # is very cumbersome compared to using the release tarball, where the
+  # `./bootstrap` step was already executed and we can `./configure` right
+  # away.
+  src = fetchurl {
+    url = "https://github.com/limine-bootloader/limine/releases/download/v${version}/limine-${version}.tar.gz";
+    sha256 = "sha256-oaQA461MGVuG7aJri2HyYb6Pu8BXOWTF3Wq/SputQSw=";
+  };
+
+  nativeBuildInputs = [
+    llvmPackages.bintools
+    llvmPackages.lld
+    mtools
+    nasm
+  ];
+
+  configureFlags = [
+    "TOOLCHAIN_FOR_TARGET=llvm"
+    "--enable-all"
+  ];
+
+  installFlags = [ "destdir=$out" "manprefix=/share" ];
+
+  outputs = [ "out" "doc" "dev" "man" ];
```

Comment:
Are these `installFlags` and `outputs` options really necessary? Judging from looking at other autotools packages, they do not seem to be, and they seem to potentially be in error.

Reply:
Hmm okay. But *why* are they necessary over letting Nix do its thing? Is this something that changed compared to some older Nix version? What happens differently if these 2 assignments are omitted?

Pardon my insistence, I am asking because I have checked against several autotools packages already packaged in nixpkgs and they don't have this.

Reply:
I hope this helps @mintsuki 

![image](https://github.com/NixOS/nixpkgs/assets/5737016/662dfc3b-5d5b-437a-90ff-44b8ef477b17)

Reply:
Alright, okay about the `outputs` line, but what about `installFlags`? Limine's build system never uses `destdir` nor `manprefix` variables... are you sure this is doing anything? The only precedent I can find for this is in the `pkgs/os-specific/linux/mceinject/default.nix` package, but in that case the plain makefile mce-inject uses does use `destdir` and `manprefix`.

Reply:
I have that from @blitz - Can you help out?

Reply:
Okay, `installFlags` can be dropped. The Nix magic still work. But we need `outputs` tho to get everything right.